src/python/pants/backend/python/target_types_rules.py

```diff
@@ -272,12 +273,47 @@ async def infer_pex_binary_entry_point_dependency(
     maybe_disambiguated = explicitly_provided_deps.disambiguated(
         owners.ambiguous, owners_must_be_ancestors=entry_point.file_name_used
     )
-    unambiguous_owners = owners.unambiguous or (
-        (maybe_disambiguated,) if maybe_disambiguated else ()
-    )
+
+    if owners.unambiguous:
+        unambiguous_owners = owners.unambiguous
+    elif maybe_disambiguated:
+        unambiguous_owners = (maybe_disambiguated,)
+    elif owners.ambiguous and not maybe_disambiguated:
+        _handle_unresolved_pex_entrypoint(
+            address,
+            entry_point_field.value,
+            owners.ambiguous,
+            python_infer_subsystem.unowned_dependency_behavior,
+        )
+        unambiguous_owners = ()
+    else:
+        unambiguous_owners = ()
     return InferredDependencies(unambiguous_owners)
 
 
+def _handle_unresolved_pex_entrypoint(
+    address: Address,
+    entry_point: str,
+    ambiguous_owners,
+    unowned_dependency_behavior: UnownedDependencyUsage,
+):
+    """Raise an error if we could not disambiguate an entrypoint for the PEX."""
+    msg = softwrap(
+        f"""
+        Pants cannot resolve the entrypoint for the target {address}.
+        The entrypoint {entry_point} might refer to the following:
+
+        {bullet_list(o.spec for o in ambiguous_owners)}
+        """
+    )
+    if unowned_dependency_behavior is UnownedDependencyUsage.DoNothing:
+        return
+    elif unowned_dependency_behavior is UnownedDependencyUsage.LogWarning:
+        logger.warning(msg)
+    else:
+        raise UnownedDependencyError(msg)
+
```

Comment:
Also personal preference:
(it's obvious there is no more code after this one. Especially since it raises)

```suggestion
    raise UnownedDependencyError(msg)

```

Reply:
I like to include the `else` case when the chain is on the value of an enum, helps me read it as a `match` block